### PR TITLE
Add analyze and format GitHub action

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
To ensure that changes are not merged to main (or feature branches) that do not lint, or are not formatted correctly.

## Changes
- Added a GitHub action that will run `flutter analyze` and `dart format` with the requirement that no issues are found.
- Commented out `print` commands that `flutter analyze` disallows (later I would just have deleted them, but perhaps we still need them for testing).
- Added `.vscode` to gitignore as that holds the local configuration of the custom line width of 150 when using vscode.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Review by bisgardo — APPROVED** [review]
Looks good. Once merged, we can add the workflow as required in the repo settings.

**Comment by abizjak** [comment]
> Looks good. Once merged, we can add the workflow as required in the repo settings.

Github Actions has problems and will sometimes prevent PRs from merging if a workflow that is conditional, and only runs on certain files changed, has not run because nothing relevant has changed. It will still prevent merging.

This is why we typically don't prevent merging PRs unless all jobs have completed. But rather just discourage by showing them as red. It works because people are not malicious.

**Comment by bisgardo** [comment]
> > Looks good. Once merged, we can add the workflow as required in the repo settings.
> 
> Github Actions has problems and will sometimes prevent PRs from merging if a workflow that is conditional, and only runs on certain files changed, has not run because nothing relevant has changed. It will still prevent merging.
> 
> This is why we typically don't prevent merging PRs unless all jobs have completed. But rather just discourage by showing them as red. It works because people are not malicious.

We do have this enabled on other repos. Not denying that these things happen, but I don't recall being blocked by it. It's a short job and you can always restart it if something isn't working. And bypass the protection if a broken workflow (runner) is all that's preventing the merge.

I suggest we add it, we can always remove it again.

**Pull request merged**
